nemo/core/classes/modelPT.py

```diff
@@ -177,12 +178,39 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
                     f"Test config : \n{OmegaConf.to_yaml(self._cfg.test_ds)}"
                 )
 
+        # Create list of lists for val and test outputs to support multiple dataloaders
+        self.validation_step_outputs = (
+            []
+        )  # Initialize an empty list as sometimes self._validation_dl can be None at this stage
+        # Check len(self._validation_dl) > 1 as sometimes single dataloader can be in a list: [<Dataloader obj>] when ds_item in
+        # config has 1 item passed in a list
+        if self._validation_dl and type(self._validation_dl) == list and len(self._validation_dl) > 1:
+            for _ in range(len(self._validation_dl)):
+                self.validation_step_outputs.append([])
+
+        self.test_step_outputs = []  # Initialize an empty list as sometimes self._test_dl can be None at this stage
+        if self._test_dl and type(self._test_dl) == list and len(self._test_dl) > 1:
+            for _ in range(len(self._test_dl)):
+                self.test_step_outputs.append([])
         # ModelPT wrappers over subclass implementations
         self.training_step = model_utils.wrap_training_step(self.training_step)
 
         # Setup nsys profiling if it has been enabled in the model config
         self._setup_nsys_profiling()
 
+    def _prefetch(self, iterator):
```

Comment:
Hmm does NeMo core really need to provide this? This seems to be more of a data thing than a NeMo core thing. I see its used in Megatron, maybe best to move it to NLPModel level if its not used anywhere else.